src/index/base.cpp

```diff
@@ -91,11 +91,12 @@ bool BaseIndex::Init()
             const CBlockIndex* block = active_chain.Tip();
             prune_violation = true;
             // check backwards from the tip if we have all block data until we reach the indexes bestblock
-            while (block_to_test && block->pprev && (block->pprev->nStatus & BLOCK_HAVE_DATA)) {
+            while (block_to_test && block && (block->nStatus & BLOCK_HAVE_DATA)) {
                 if (block_to_test == block) {
                     prune_violation = false;
                     break;
                 }
+                assert(block->pprev);
```

Comment:
It may be worth adding a comment here on _why_ `block->pprev` must exist (because we're walking backwards from the active chain tip to a block index in the active chain. If a block index that we encounter doesn't have a backwards pointer, then the chain is malformed).

Reply:
This isn't actually a correct explanation (the genesis block is in the active chain and doesn't have `pprev` set). The reason this assertion will hold is because of how `block` and `block_to_test` are chosen; the conditional above this line is guaranteed to break at-or-higher-than genesis because of the few lines of logic above (which, if summarized in a comment, would just end up repeating the fairly-straightforward code above).

Reply:
I think you're saying the same thing as I am. Because of the way `block` and `block_to_test` are chosen, if we walk backwards from `block`, we're guaranteed to encounter `block_to_test` (either at or before the genesis block).

Reply:
Sorry I didn't get to this earlier, would have added a comment, do people think that this warrants a follow-up?

Reply:
@mzumsande sounds like it could be.

Reply:
done in #23777